### PR TITLE
COMMERCE-9376 Save as Draft to save as a new version of the Product

### PR DESCRIPTION
**Pull request by pietrobocale**
https://issues.liferay.com/browse/COMMERCE-9376

### CODE REVIEW

**Comment by pietrobocale** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
ci:forward

**Comment by pietrobocale** [comment]
ci:stop

**Comment by pietrobocale** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
Base PR: https://github.com/liferay-commerce/liferay-portal/pull/2790

File: modules/apps/commerce/commerce-product-test/src/testFunctional/paths/CommerceEntry.path [hunk 1]
```diff
@@ -987,6 +987,11 @@
 	<td>xpath=((//strong[contains(.,'${key_status}')])[${key_row}]/../../../..//a[text()='${key_productName}'])</td>
 	<td></td>
 </tr>
+<tr>
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/paths/CommerceEntry.path:990]: Please group paths on the same commit

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProducts.macro [hunk 2]
```diff
@@ -361,4 +361,41 @@ definition {
 			}
         }
 	}
+
+	macro checkProductsNotPresent { 
+		for (var value : list "${list}") {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProducts.macro]: Same here for the indentation

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProducts.macro [hunk 3]
```diff
@@ -398,4 +398,36 @@ definition {
 			}
         }
 	}
+
+	macro checkProductId { 
+		for (var value : list "${list}") {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProducts.macro]: Same here for the indentation

File: modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProducts.macro [hunk 4]
```diff
@@ -430,4 +430,61 @@ definition {
 
         } 
 	}
+
+	macro checkProductDescription {			
+		for (var value : list "${list}") {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/macros/CommerceProducts.macro]: Same here for the indentation

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase [hunk 5]
```diff
@@ -16,4 +16,30 @@ definition {
...
+	@priority = "3"
+	test CanCheckVersionableProductSaveAsDraftButton {
+        property portal.acceptance = "false";
+
+        task ("When the Admin goes into the Commerce -> Products and create a new simple product") {
+            CommerceNavigator.gotoPortlet(
+			    category = "Product Management",
+			    portlet = "Products");
+
+		    CPCommerceCatalog.newProductsEntry(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase]: Check the indentation here

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase [hunk 6]
```diff
@@ -310,4 +310,95 @@ definition {
...
+        property test.name.skip.portal.instance = "CPCommerceProductVersioning#CanCheckProductVersionsWithProductVersioningDisabled";
+
+        //This test need to disable the versioning for that reason the 'skip.portal.instance' is necessary
+
+        task ("Given the admin has already created a new simple product and When the admin opens the product that was created before") {
+            CommerceNavigator.gotoPortlet(
+			    category = "Product Management",
+			    portlet = "Products");
+
+		    CPCommerceCatalog.newProductsEntry(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase]: Check the indentation here

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase [hunk 7]
```diff
@@ -401,4 +401,74 @@ definition {
...
+	@priority = "3"
+	test CanCheckVersionableProductConvertToDraftStatus {
+        property portal.acceptance = "false"; 
+         
+        task ("Given the admin has already created a new simple product and When the admin opens the product that was created before") {
+            CommerceNavigator.gotoPortlet(
+			    category = "Product Management",
+			    portlet = "Products");
+
+		    CPCommerceCatalog.newProductsEntry(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase]: Check the indentation here

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase [hunk 8]
```diff
@@ -401,4 +401,74 @@ definition {
...
+            Navigator.gotoBack();
+
+            Click(
+			    key_productName = "Simple T-Shirt",
+			    key_status = "Approved",
+			    locator1 = "CommerceEntry#PRODUCT_ENTRY_BASED_ON_STATUS");
+        }
+
+        task ("And click on 'Convert To Draft' (save the ID, the ID should remain the same for this product)") {
+			Click(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase:398]: Same here for the indentation

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase [hunk 9]
```diff
@@ -401,4 +401,74 @@ definition {
...
+            AssertElementPresent(
+			    key_status = "Draft",
+			    locator1 = "CommerceEntry#WORKFLOW_STATUS");
+
+        }
+
+        task ("Then the admin will see 2 products, with different id and into the Draft status") {
+            CommerceProducts.checkProductIdsEqual(
+                firstProductId = "${approvedProductId}",
+				secondProductId = selenium.getText("CommerceEntry#PRODUCT_ID"),
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductVersioning.testcase]: Same here for the indentation

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@pietrobocale please check my comment. Move the autogenerated commits at the end. Thanks!

**Comment by alessiorendina** [comment]
The autogenerated commits to move are:

**COMMERCE-9376 Build lang
COMMERCE-9376 Regen services
COMMERCE-9376 Semantic versioning (gw baseline)**

**Comment by pietrobocale** [comment]
ci:forward

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
@pietrobocale please move the commits containing autogenerated files at the end, thanks.

**Comment by alessiorendina** [comment]
ci:stop

**Comment by pietrobocale** [comment]
ci:test:sf

**Comment by pietrobocale** [comment]
ci:forward

**Comment by alessiorendina** [comment]
@pietrobocale can you run baseline and fixup last commit? Thanks

**Comment by pietrobocale** [comment]
ci:forward

**Pull request closed without merging**